validator/accounts/v2/accounts_deposit_test.go

```diff
@@ -0,0 +1,220 @@
+		assert.NoError(tb, set.Set(flags.Eth1KeystorePasswordFileFlag.Name, cfg.eth1KeystorePasswordFile))
+	}
+	if cfg.eth1PrivateKeyFile != "" {
+		assert.NoError(tb, set.Set(flags.Eth1PrivateKeyFileFlag.Name, cfg.eth1PrivateKeyFile))
+	}
+	if cfg.publicKeysFlag != "" {
+		assert.NoError(tb, set.Set(flags.DepositPublicKeysFlag.Name, cfg.publicKeysFlag))
+	}
+	if cfg.depositAllAccountsFlag == true {
+		assert.NoError(tb, set.Set(flags.DepositAllAccountsFlag.Name, strconv.FormatBool(cfg.depositAllAccountsFlag)))
+	}
+	assert.NoError(tb, set.Set(flags.SkipDepositConfirmationFlag.Name, strconv.FormatBool(cfg.skipDepositConfirmationFlag)))
+	return cli.NewContext(&app, set, nil)
+}
+
+func TestCreateDepositConfig(t *testing.T) {
+	walletDir, _, passwordFilePath := setupWalletAndPasswordsDir(t)
+
+	// First, create the wallet and several accounts
+	cliCtx := setupWalletCtx(t, &testWalletConfig{
+		keymanagerKind:     v2keymanager.Derived,
+		walletDir:          walletDir,
+		walletPasswordFile: passwordFilePath,
+		skipDepositConfirm: true,
+	})
+	wallet, err := CreateAndSaveWalletCli(cliCtx)
+	require.NoError(t, err)
+
+	err = CreateAccount(cliCtx.Context, &CreateAccountConfig{
+		Wallet:      wallet,
+		NumAccounts: 3,
+	})
+	require.NoError(t, err)
+	keymanager, err := wallet.InitializeKeymanager(
+		cliCtx.Context,
+		true, /* skip mnemonic confirm */
+	)
+	require.NoError(t, err)
+
+	// Save public keys for comparison and selection purposes later
+	pubkeys, err := keymanager.FetchValidatingPublicKeys(cliCtx.Context)
+	require.NoError(t, err)
+	var hexPubkeys []string
+	for _, pubkey := range pubkeys {
+		encoded := make([]byte, hex.EncodedLen(len(pubkey)))
+		hex.Encode(encoded, pubkey[:])
+		hexPubkeys = append(hexPubkeys, string(encoded))
+	}
+	// Remove the last key so that we can select a portion and not all of the accounts later
+	hexPubkeys = hexPubkeys[:len(hexPubkeys)-1]
+	hexPubkeysString := strings.Join(hexPubkeys, ",")
+
+	// Create a file holding the test ETH1 private key
+	eth1PrivateKeyFile, err := ioutil.TempFile("", "testing")
+	require.NoError(t, err)
+	defer func() {
+		err = eth1PrivateKeyFile.Close()
+		require.NoError(t, err)
+		err = os.Remove(eth1PrivateKeyFile.Name())
+		require.NoError(t, err)
+	}()
+	_, err = eth1PrivateKeyFile.WriteString("This should be an ETH1 private key")
+	require.NoError(t, err)
+
+	// First we test the behavior when depositAllAccountsFlag is set to true
+	depositConfig, err := createDepositConfigHelper(t, &depositTestWalletConfig{
+		keymanagerKind:              v2keymanager.Derived,
+		walletDir:                   walletDir,
+		walletPasswordFile:          passwordFilePath,
+		skipDepositConfirmationFlag: true,
+		depositAllAccountsFlag:      true,
+		httpWeb3ProviderFlag:        "http://localhost:8545",
+		eth1PrivateKeyFile:          eth1PrivateKeyFile.Name(),
+	})
+	require.NoError(t, err)
+	if len(depositConfig.DepositPublicKeys) != 3 {
```

Comment:
```suggestion
	require.Equal(t, 3, len(depositConfig.DepositPublicKeys), "wrong number of public keys")
```

Same with the occurrences below

Reply:
Will address in a commit soon. Thanks!